File: src/main/java/me/mucloud/plugin/MK/JoinMessage/PermissionMessage.java
package me.mucloud.plugin.MK.JoinMessage;

public class PermissionMessage {

    private final String Permission;
    private final int Priority;
    private final String JoinMessage;
    private final String ExitMessage;

    protected PermissionMessage(String permission,int priority,String joinmessage,String exitmessage){
        Permission = permission;
        Priority = priority;
        JoinMessage = joinmessage;
        ExitMessage = exitmessage;
    }

    public String getJoinMessage(){
        return JoinMessage;
    }

    public String getExitMessage(){
        return ExitMessage;
    }

    public String getPermission(){
        return Permission;
    }

    public int getPriority(){
        return Priority;
    }

}

File: src/main/kotlin/me/mucloud/plugins/MK/JoinMessage/PermissionMessage.kt
package me.mucloud.plugins.MK.JoinMessage

import org.bukkit.ChatColor

class PermissionMessage constructor(

    permission: String,
    priority: Int,
    joinMessage: String?,
    exitMessage: String?

){

    private var Permission: String = permission
    private var Priority: Int = priority
    private var JoinMessage: String? = joinMessage
    private var ExitMessage: String? = exitMessage

    fun getPermission(): String{
        return Permission
    }

    fun getPriority(): Int{
        return Priority
    }

    fun getJoinMessage(): String{
        return changeMessage(JoinMessage)
    }

    fun getExitMessage(): String{
        return changeMessage(ExitMessage)
    }

    private fun changeMessage(message: String?): String{
        return ChatColor.translateAlternateColorCodes('&', message.toString())
    }

}